SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs

```diff
@@ -927,6 +927,38 @@ public PersonalRelationshipType GetPersonalRelationshipTypeByDescription(string
                 .FirstOrDefault(prt => prt.Description.ToLower() == description.ToLower());
         }
 
+        public Infrastructure.PersonalRelationship GetPersonalRelationshipById(long relationshipId)
+        {
+            return _databaseContext.PersonalRelationships
+                .FirstOrDefault(prt => prt.Id == relationshipId);
+        }
+
+        public void DeleteRelationships(Infrastructure.PersonalRelationship relationship)
+        {
+            var type = _databaseContext.PersonalRelationshipTypes.FirstOrDefault(tp => tp.Id == relationship.TypeId);
+            var inverseType = _databaseContext.PersonalRelationshipTypes.FirstOrDefault(tp => tp.Id == type.InverseTypeId);
+
+            var secondRelationship = _databaseContext.PersonalRelationships
+                .FirstOrDefault(prt => prt.PersonId == relationship.OtherPersonId && prt.TypeId == inverseType.Id);
+
+            var relationshipDetails = _databaseContext.PersonalRelationshipDetails.FirstOrDefault(prd => prd.PersonalRelationshipId == relationship.Id);
+            var secondRelationshipDetails = _databaseContext.PersonalRelationshipDetails.FirstOrDefault(prd => prd.PersonalRelationshipId == secondRelationship.Id);
+
+            if (relationshipDetails != null)
+            {
+                _databaseContext.PersonalRelationshipDetails.Remove(relationshipDetails);
+            }
+            if (secondRelationshipDetails != null)
+            {
+                _databaseContext.PersonalRelationshipDetails.Remove(secondRelationshipDetails);
+            }
```

Comment:
Have we had a look to see if EF Core can help us out a bit here? Looks like we might be about to `Include` `PersonalRelationshipDetail` and then remove, that way we don't have to do two finds for them. https://docs.microsoft.com/en-us/ef/core/saving/cascade-delete#deleting-a-principalparent